components/buttons.js

```diff
@@ -1,10 +1,11 @@
 import { TouchableOpacity, Text } from "react-native";
 import { useTailwind } from "tailwind-rn";
 
-const RoundButton = ({ style, title }) => {
+const RoundButton = ({ style, title, singUpAuth }) => {
```

Comment:
Given that `RoundButton` is a general button that is meant to be used all over the application, not necessarily just in the `signUp` page, we should not add a property specific for authentication here. Rather we should make it more general. Here is an idea of how we can accomplish this:

in buttons.js

```jsx
const RoundButton = ({ style, title, onPress }) => {

  // then later on
  return (
    <TouchableOpacity
      onPress = { () => onPress() }
      style={[
        tailwind("flex justify-center items-center rounded-full p-4"),
        style,
      ]}
    >
      <Text style={tailwind("text-white text-xl font-bold")}>{title}</Text>
    </TouchableOpacity>
  );
}
```

Now we can pass anything we want to the `onPress()` function

Example in singnUp.js

```jsx
<RoundButton style={tailwind("bg-green-500 mb-2")} onPress= { () => {// do something here} }  
title="sign up" />
```

Reply:
I'll remove the authentication process from the button, I may have looked too ahead of myself here and prepare for authentication. I commented out authentication things from App.js as well